trie-db/src/triedbmut.rs

```diff
@@ -2099,24 +2360,16 @@ where
 			None => {
 				#[cfg(feature = "std")]
 				trace!(target: "trie", "remove: obliterated trie");
-				self.root_handle = NodeHandle::Hash(L::Codec::hashed_null_node());
-				*self.root = L::Codec::hashed_null_node();
+				self.root_handle =
+					NodeHandle::Hash(L::Codec::hashed_null_node(), Default::default());
+				self.root = L::Codec::hashed_null_node();
 			},
 		}
 
 		Ok(old_val)
 	}
 }
 
-impl<'a, L> Drop for TrieDBMut<'a, L>
-where
-	L: TrieLayout,
-{
-	fn drop(&mut self) {
-		self.commit();
-	}
-}
```

Comment:
agree with the removal, this action should be implicit I think. TODO Maybe worth putting in release note.

Reply:
`self.db` is read-only anyway, so drop can't really do anything.
There's a lot of changes in the API, hence the new crate.